packages/database/src/TupaiaDatabase.js

```diff
@@ -465,12 +465,22 @@ function buildQuery(connection, queryConfig, where = {}, options = {}) {
     options.columns.map(columnSpec => {
       if (typeof columnSpec === 'string') return columnSpec;
       const [alias, selector] = Object.entries(columnSpec)[0];
+
+      if (selector?.castAs) {
+        const { castAs } = selector;
+        if (!VALID_CAST_TYPES.includes(castAs)) {
+          throw new Error(`Cannot cast as ${castAs}`);
+        }
+        return { [alias]: connection.raw(`??::${castAs}`, [alias]) };
+      }
```

Comment:
@edmofro you may want to have a look at the changes in this file since you are the one with the best knowledge of the core DB classes

Reply:
Looks good!